src/octoprint/plugins/classicwebcam/__init__.py

```diff
@@ -0,0 +1,261 @@
+                "suffix": "_mock",
+            },
+            {
+                "type": "wizard",
+                "name": "Classic Webcam Wizard",
+                "template": "classicwebcam_wizard.jinja2",
+                "suffix": "_wizard",
+            },
+        ]
+
+    # ~~ WebcamPlugin API
+
+    def get_webcam_configurations(self):
+        streamRatio = self._settings.get(["streamRatio"])
+        if streamRatio == "4:3":
+            streamRatio = RatioEnum.four_three
+        else:
+            streamRatio = RatioEnum.sixteen_nine
+        webRtcServers = self._settings.get(["streamWebrtcIceServers"])
+        cacheBuster = self._settings.get_boolean(["cacheBuster"]) is True
+        stream = self._get_stream_url()
+        snapshot = self._get_snapshot_url()
+        flipH = self._settings.get_boolean(["flipH"]) is True
+        flipV = self._settings.get_boolean(["flipV"]) is True
+        rotate90 = self._settings.get_boolean(["rotate90"]) is True
+        snapshotSslValidation = (
+            self._settings.get_boolean(["snapshotSslValidation"]) is True
+        )
+
+        try:
+            streamTimeout = int(self._settings.get(["streamTimeout"]))
+        except Exception:
+            streamTimeout = 5
+
+        try:
+            snapshotTimeout = int(self._settings.get(["snapshotTimeout"]))
+        except Exception:
+            snapshotTimeout = 5
+
+        return [
+            Webcam(
+                name="classic",
+                displayName="Classic Webcam",
+                flipH=flipH,
+                flipV=flipV,
+                rotate90=rotate90,
+                snapshotDisplay=snapshot,
+                canSnapshot=self._can_snapshot(),
+                compat=WebcamCompatibility(
+                    stream=stream,
+                    streamTimeout=streamTimeout,
+                    streamRatio=streamRatio,
+                    cacheBuster=cacheBuster,
+                    streamWebrtcIceServers=webRtcServers,
+                    snapshot=snapshot,
+                    snapshotTimeout=snapshotTimeout,
+                    snapshotSslValidation=snapshotSslValidation,
+                ),
+                extras=dict(
+                    stream=stream,
+                    streamTimeout=streamTimeout,
+                    streamRatio=streamRatio,
+                    streamWebrtcIceServers=webRtcServers,
+                    cacheBuster=cacheBuster,
+                ),
+            ),
+        ]
+
+    def _get_snapshot_url(self):
+        return self._settings.get(["snapshot"])
+
+    def _get_stream_url(self):
+        return self._settings.get(["stream"])
+
+    def _can_snapshot(self):
+        snapshot = self._get_snapshot_url()
+        return snapshot is not None and snapshot.strip() != ""
+
+    def take_snapshot(self, _):
+        snapshot_url = self._get_snapshot_url()
+        if self._can_snapshot() is False:
+            raise Exception("Snapshot is not configured")
```

Comment:
Should be a special Exception type. Should live in `octoprint.webcams` I think.

Reply:
Now raising `WebcamNotAbleToTakeSnapshotException` with proper message